### PR TITLE
Feat / add slide boundary when using cycle mode stop

### PR DESCRIPTION
**Pull request by ChristiaanScheermeijer**
We have support for three "cycle modes" in the TileSlider library, but we didn't implement full support for the cycle mode stop yet, until now! 

**The issues found:**

- Swiping the slider causes an empty space
- Sliding with the keys or API causes this as well
- The velocity animation wasn't stopped when the boundary is reached

Here is how it works now:

https://github.com/user-attachments/assets/c599cb28-4bb0-4063-b64d-35538fe12b80

This PR addresses these issues and adds an example page with cycle mode set to stop. I refactored some of the drag clamping functions because these were repeated and both needed account for the cycle mode stop boundaries.

### CODE REVIEW

File: src/TileSlider.tsx [hunk 1]
```diff
@@ -136,20 +136,26 @@ export const TileSlider = <T,>({
     frameWidth: 0,
   });
 
-  const calculateIndex = useCallback(() => {
-    const tileWidth = sliderDataRef.current.frameWidth / tilesToShow;
-    let index = Math.round((sliderDataRef.current.position / tileWidth) * -1);
+  const getIndexWithBoundary = useCallback(
```
Comment (royschut) [inline src/TileSlider.tsx]: Minor: it doesn't `get` the index, it just limits it: `limitIndex()`, `indexWithBoundaries()`

File: src/TileSlider.tsx [hunk 2]
```diff
@@ -275,6 +282,13 @@ export const TileSlider = <T,>({
       // total duration of the snap animation from the startTime
       const totalDurationSnap = snappingStartTime + snappingDuration - startTime;
 
+      // in cycle mode stop, quickly cancel out the velocity animation when we have reached a boundary
+      if (cycleMode === CYCLE_MODE_STOP && !boundaryReached && (currentIndex <= 0 || currentIndex >= totalTiles - tilesToShow)) {
+        startTime = Date.now();
```
Comment (royschut) [inline src/TileSlider.tsx]: Smart. Perhaps add a comment why we reset the time?

**Review by royschut — APPROVED** [review]
Nice work!

File: src/TileSlider.tsx (hunk 1 above)
Comment (ChristiaanScheermeijer) [reply src/TileSlider.tsx]: That's right, I will update the name!

**Comment by langemike** [comment]
Great work @ChristiaanScheermeijer! And thanks for helping me out.

I noticed two issues, though. It will take me some time to point you to the root cause in the code, but hopefully you see it a lot quicker than me.

I can only reproduce these issues within my implementation and not on your demo-page. I can also reproduce it in "Responsive mode" of my desktop browser.

1. When you reached the end of a slider with MANY items, when you perform the last swipe it will go back to the beginning
2. When you reached the end of slider with 3 items and you perform the last swipe, it will go one item back.

These issues doesn't occur at the start, only at the end.

You will see these issues in my screen recording. In my implementation I did nothing weird, so I suspect it being a bug in the calculation.

https://github.com/user-attachments/assets/51493d27-edb0-48e1-8e02-0f87d01eb515

**Comment by royschut** [comment]
> I noticed two issues, though.

How about fixing this separately? This PR makes it a lot better, so then we can start using it. Today. :-)

**Comment by ChristiaanScheermeijer** [comment]
Thanks @langemike, I think I also found this issue caused by 2 things;

- I was using `totalTiles` which represents the tiles rendered including the overscan. This was, coincidentally, exactly the amount used in my tests...
- The snap transition was a bit weird and I believe it is not needed. It can still be improved to visually see the edge snapping, but we can do this later!

I'm curious if you still see these issues when using the latest commit.

**Comment by langemike** [comment]
> How about fixing this separately? This PR makes it a lot better, so then we can start using it. Today. :-)

Respectfully, I rather use the `endless` mode than this. This is similarly bad but in a whole different way.

EDIT: this comment was placed at the same time christiaan wrote his previous comment. I will retest with the latest commit.

**Comment by langemike** [comment]
> I'm curious if you still see these issues when using the latest commit.

This is PERFECT. It works really well, thanks for the fix!

**Review by langemike — APPROVED** [review]

**Pull request merged**